src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/BuildUpToDateCheck.cs

```diff
@@ -166,6 +158,27 @@ private void OnProjectChanged(IProjectSubscriptionUpdate e)
                 _customOutputs.AddRange(outputs.After.Items.Select(item => item.Value[UpToDateCheckOutput.FullPathProperty]));
             }
 
+            if (e.ProjectChanges.TryGetValue(UpToDateCheckBuilt.SchemaName, out var built) &&
+                built.Difference.AnyChanges)
+            {
+                _builtOutputs.Clear();
+
+                foreach (var item in built.After.Items)
+                {
+                    var destination = item.Value[UpToDateCheckBuilt.IdentityProperty];
+
+                    if (item.Value.TryGetValue(UpToDateCheckBuilt.OriginalProperty, out var source) &&
+                        !string.IsNullOrEmpty(source))
+                    {
+                        _copiedOutputFiles[destination] = source;
```

Comment:
Any value in extending this to support N:1 inputs:outputs? That'd be nice for, say, a compiler, but I don't know how important the scenario of someone smuggling an additional compiler into their project is.

Reply:
Yeah, I thought about that when I was doing it. I figured we'd cross the bridge when we came to it by supporting a semicolon list, but if there'd be a better way to do it, we could be more general.

---
In reply to: [161843672](https://github.com/dotnet/project-system/pull/3142#discussion_r161843672) [](ancestors = 161843672)